crates/matrix-sdk/src/sliding_sync.rs

```diff
@@ -290,6 +296,91 @@ impl SlidingSyncBuilder {
         self.views = Some(views);
         self
     }
+
+    /// Activate e2ee, to-device-message and account data extensions if not yet
+    /// configured.
+    ///
+    /// Will leave any extension configuration found untouched, so the order
+    /// does not matter.
+    pub fn with_common_extensions(mut self) -> Self {
+        {
+            let mut lock = self.extensions.get_or_insert_with(Default::default).lock_mut();
+            let mut cfg = lock.get_or_insert_with(Default::default);
+            if cfg.to_device.is_none() {
+                cfg.to_device = Some(assign!(ToDeviceConfig::default(), {enabled : Some(true)}));
+            }
+
+            if cfg.e2ee.is_none() {
+                cfg.e2ee = Some(assign!(E2EEConfig::default(), {enabled : Some(true)}));
+            }
+
+            if cfg.account_data.is_none() {
+                cfg.account_data =
+                    Some(assign!(AccountDataConfig::default(), {enabled : Some(true)}));
+            }
+        }
```

Comment:
I don't see why this block is necessary.

```suggestion
        let mut lock = self.extensions.get_or_insert_with(Default::default).lock_mut();
        let mut cfg = lock.get_or_insert_with(Default::default);

        if cfg.to_device.is_none() {
            cfg.to_device = Some(assign!(ToDeviceConfig::default(), {enabled : Some(true)}));
        }

        if cfg.e2ee.is_none() {
            cfg.e2ee = Some(assign!(E2EEConfig::default(), {enabled : Some(true)}));
        }

        if cfg.account_data.is_none() {
            cfg.account_data =
                Some(assign!(AccountDataConfig::default(), {enabled : Some(true)}));
        }
```

Reply:
iirc, because `self.extensions.get_or_insert_with(Default::default).lock_mut()` was still holding a reference of `self`, when we try to return it later ...

Reply:
yeah. that was it:
```
error[E0505]: cannot move out of `self` because it is borrowed
   --> crates/matrix-sdk/src/sliding_sync.rs:320:9
    |
306 |         let mut lock = self.extensions.get_or_insert_with(Default::default).lock_mut();
    |                        ---------------------------------------------------- borrow of `self.extensions` occurs here
...
320 |         self
    |         ^^^^ move out of `self` occurs here
321 |     }
    |     - borrow might be used here, when `lock` is dropped and runs the `Drop` code for type `MutableLockMut`

For more information about this error, try `rustc --explain E0505`.
error: could not compile `matrix-sdk` due to previous error
```

Reply:
Could `drop()` the lock instead, but it actually has a significant downside: https://github.com/rust-lang/rust/issues/57478

Reply:
I thought the lock would have been freed before that. Let's keep the block then :-). Thanks for the explanations.

Reply:
unless someone has a hard want on this, I'd keep the block for the time being..